### ISSUE TITLE
Sharing: Default Pinterest method?

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#2650]
We've gone back and forth a bit on what the Pinterest sharing buttons do—does it load the multi-image select modal or choose a single image for a Pinner to use. There has been inconsistency on what the official vs unofficial buttons do.

Currently, we have a filter in place that can choose between the two options and defaults to the single image option.

I think the multiimage modal provides more flexibility to both the site owner and the user (e.g. blogs that include multiple "Pinterest-friendly" images in the post body).

Thoughts on changing the default to the multi-image modal?

@aduth @csonnek

**Comment by macmanx2** [comment]
Pinterest is about sharing images, so I strongly believe that the default should be the multi-image modal.

**Comment by csonnek** [comment]
I would prefer to have the multi-image be the default for our buttons. While the Pinterest official button we use only shares the single image - it can be <a href="https://help.pinterest.com/en/articles/build-pin-it-button-your-site">configured</a> for multiple images. 

In addition, their official browser <a href="https://help.pinterest.com/en/articles/add-pinterest-browser-button#Web">bookmarklet</a> defaults to multi-image.

**Comment by aduth** [comment]
With multi-image mode enabled, what would be the expected outcome of clicking the "Pin It" button below a post when in the context of a blog list page? Personally, I'd expect images to relate only to that particular post, but I seem to recall there being challenges with achieving this effect and that, instead, it would show all images found on the blog listing page.

**Comment by csonnek** [comment]
As an avid Pinterest user, most sites that offer Pinterest sharing will show all the blog images as options for pinning. I think our button should follow that same behavior.

**Comment by kraftbj** [comment]
@aduth Yeah, it does offer to select from all images visible on an archive page, not from a specific post within a page. The Pinterest browser extension does the same behavior when pinning an archive page.

**Comment by s-a-s-k-i-a** [comment]
Am I the only one having problems successfully using the Pinterest Sharing Button?
I disabled it on my site, because it spit out some bogus URL and sent my users to some 404 causing site on my domain.

The link that is being addressed when clicking Pinterest-Sharing-Button looks like this on my site:
https://www.saskialund.de/yoast-seo-plugin-optimal-einstellen-so-gehts/?share=pinterest&nb=1

This is where it redirects me after clicking Jetpacks Pinterest-Sharing Button:
https://www.saskialund.de/www.pinterest.com/pin/create/button/?url=https%3A%2F%2Fwww.saskialund.de%2Fyoast-seo-plugin-optimal-einstellen-so-gehts%2F&media=https%3A%2F%2Fwww.saskialund.de%2Fwp-content%2Fuploads%2Fyoast-seo-post-2.jpg&description=Yoast%20SEO%20Plugin%20optimal%20einstellen%3A%20so%20geht%E2%80%99s%21
and this leads to a 404 page...

**Comment by kraftbj** [comment]
@s-a-s-k-i-a That issue is #2838, resolved in #2840, which will be in the next version of Jetpack.

**Comment by s-a-s-k-i-a** [comment]
@kraftbj Hi and thanks! I was looking at exactly that code portion just now, when your essage arrived and fixed it by adding the https protocol instead of the relative protocol double slash only version :)

Thanks once more!

**Comment by annezazuu** [comment]
2697269-t multiple images requested

**Comment by chaselivingston** [comment]
2863164-t

**Comment by jenhooks** [comment]
2875019-t

**Comment by jeherve** [comment]
From #9804:

------

using the Pinterest share button used to allow the sharer to choose the image to pin, in a way similar to this: https://cloudup.com/cOTwU_x2c0n Now, the Pinterest share button forces the sharer to use the _featured_ image of the post. 

>If "official buttons" setting is set for the sharing buttons, each of the images on the post will have its own pin button on hover. This is not the case with any setting other than "official button".

This doesn't address this issue of the sharing button at the bottom of the post allowing the sharer to choose the image. Readers read to the end of a post, and then they click the Pinterest sharing button. It is not normal user behavior to scroll back up to check if there's a hover button on an image.

>For any setting, using the pin button at the bottom of the post will be "smart" and grab the first/best image for the pin.

Pinterest users don't like posting small, horizontal images like the one that's typically in the post's featured image. They want to post long, vertical images, because Pinterest's algorithm scores higher when these sized images are used. For this reason, the Pinterest share button is not being "smart". Using the featured image for other social networks may make sense, but it does not make sense for Pinterest.

>If you would like to override this default, you can do so via the jetpack_sharing_pinterest_widget_type filter.

This sounds like a good solution, but it is not viable for WordPress.com simple sites. Can we implement this on WPcom?

**Comment by erricgunawan** [comment]
Have a user with a similar issue on 8072446-hc / 1571495-zen

Since they're in the Business plan, I gave them the jetpack filter workaround.

**Comment by macmanx2** [comment]
Also reported at https://wordpress.org/support/topic/only-featured-image-is-displayed-when-using-pinterest-share-button-in-post/

**Comment by keilabl** [comment]
Having the same issue and the workaround that I found is apparently no longer working:
`add_filter( 'jetpack_sharing_pinterest_widget_type', 'jetpack_pinterest_share_multi_image' );
function jetpack_pinterest_share_multi_image() {
    return 'buttonBookmark';
}`

**Comment by keilabl** [comment]
> 
> 
> @keilabl This will only work if you use the "Official" button style under Settings > Sharing in your dashboard.
> 
> The `buttonBookmark` parameter should still work as it's documented here:
> https://developers.pinterest.com/tools/widget-builder/?type=pinit
> 
> If you do, and if the workaround still doesn't work, that's something that will probably need to be reported to Pinterest directly here:
> https://github.com/pinterest/widgets/issues?q=is%3Aissue+sort%3Aupdated-desc+buttonBookmark+is%3Aopen

Actually. I had "official buttons" selected... but I just now switched off of 'Official Buttons' and selected "Icons only" instead and it's working as expected... which is backwards, but at least it's working. Thanks.

**Comment by jeherve** [comment]
Interesting. When using non-official buttons, you can remove that code snippet from your site; it will not make any changes to your sharing buttons when using non-sharing buttons.

**Comment by joweber123** [comment]
Another report here of this being confusing for a user on a WPcom simple site 3356134-zen
